src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java

```diff
@@ -0,0 +1,157 @@
+	private ProductCatalogueService productCatalogueService;
+
+	@Autowired
+	private MultimediaService multimediaService;
+
+	private static Logger logger = LoggerFactory.getLogger(ProductCatalogueResource.class.toString());
+
+	@GetMapping(produces = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<List<ProductCatalogue>> getAll(
+			@RequestParam(value = "productName", defaultValue = "", required = false) String productName
+			, @RequestParam(value = "uniqueId", defaultValue = "0", required = false) Long uniqueId,
+			@RequestParam(value = "serverVersion", required = false) String serverVersion) {
+
+		Long lastSyncedServerVersion = null;
+		if (serverVersion != null) {
+			lastSyncedServerVersion = Long.parseLong(serverVersion) + 1;
+		}
+
+		ProductCatalogueSearchBean productCatalogueSearchBean = new ProductCatalogueSearchBean();
+		productCatalogueSearchBean.setProductName(productName);
+		productCatalogueSearchBean.setUniqueId(uniqueId);
+		productCatalogueSearchBean.setServerVersion(lastSyncedServerVersion);
+
+		return new ResponseEntity<>(productCatalogueService.getProductCatalogues(productCatalogueSearchBean),
+				RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
+	}
+
+	@PostMapping(headers = { "Accept=multipart/form-data" })
+	public ResponseEntity<String> create(@RequestPart(required = false) MultipartFile file,
+			@RequestPart ProductCatalogue productCatalogue) {
+		try {
+			productCatalogueService.add(productCatalogue);
+			Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+			String userName = authentication.getName();
+			ProductCatalogue catalogue = productCatalogueService
+					.getProductCatalogueByName(productCatalogue.getProductName());
+			if (catalogue != null && file != null) {
+				Multimedia multimedia = multimediaService.findByCaseId(String.valueOf(catalogue.getUniqueId()));
+				if (multimedia == null) {
+					MultimediaDTO multimediaDTO = new MultimediaDTO(catalogue.getUniqueId().toString(), userName.trim(),
+							file.getContentType().trim(), null, "catalog_image");
+					multimediaDTO.withOriginalFileName(file.getOriginalFilename()).withDateUploaded(new Date());
+
+					logger.info("Saving multimedia file...");
+					multimediaService.saveFile(multimediaDTO, file.getBytes(), file.getOriginalFilename());
+				}
+			}
+
+			return new ResponseEntity<>(HttpStatus.CREATED);
+		}
+		catch (IOException e) {
+			logger.error(
+					String.format("Exception occurred while persisting image of Product Catalogue" + e.getMessage() + e));
+			return new ResponseEntity<String>("Exception occurred while persisting image of Product Catalogue",
+					HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		catch (IllegalArgumentException e) {
+			logger.error(String.format("Exception occurred while adding product catalogue: %s,%s", e.getMessage(), e));
+			return new ResponseEntity<String>("The request contain illegal argument : " + e.getMessage(),
+					HttpStatus.BAD_REQUEST);
+		}
```

Comment:
This should be handled by Global exception handler

Reply:
Removed